### PR TITLE
Unified dependencies, revisited

### PR DESCRIPTION
**Pull request by only-a-ptr**
The https://github.com/RigsOfRods/ror-dependencies package was created by @mikadou more than a year ago, as well as respective PR for RoR: https://github.com/RigsOfRods/rigs-of-rods/pull/924. However, I've always had build issues with it, so I've been delaying the merge.

As of now, finally, everything is in tip-top shape:
* The DirectX build issues of the dependencies were resolved (https://github.com/RigsOfRods/ror-dependencies/pull/6)
* I added explanatory error message to the CMake scripts, telling people they need to set CMAKE_PREFIX_PATH manually to find the deps on Windows.

EDIT: Fixed build error with RoRConfig - it was a bad merge of it's CMakeLists.file, it included more RoR sources than it needed.

# TODO

- [x] Add CrashRpt to the deps - we didn't use it at the time Mikadou created the package.
REMOVED: Add CrashRpt to this PR and resolve build issues, see https://github.com/RigsOfRods/ror-dependencies/pull/11#issuecomment-305191267

# Bugs
- [x] RoRConfig has a library mismatch because it uses precompiled dependencies for VS2015 instead of VS1017 
- [x] Caelum fails to load the sky scripts

### CODE REVIEW

**Comment by only-a-ptr** [comment]
@AnotherFoxGuy Apparently, AppVeyor build setup needs to be updated with the new deps.

**Comment by AnotherFoxGuy** [comment]
Fixed the AppVeyor build system, you can download the build [here](https://ci.appveyor.com/api/buildjobs/vr0ohueiiae8wc7t/artifacts/bin%2FRoR-V0.4.8.0-CIBuild-158-qpbyhcry.zip
)  
@only-a-ptr I've updated the pull request with some bugs I've found.

File: source/main/CMakeLists.txt [hunk 1]
```diff
@@ -429,39 +374,233 @@ IF(WIN32)
...
+			else()
+				message( WARNING "Could not copy ${source_file} to output directory!" )
+			endif()
+			unset( PATH_TO_SOURCE CACHE )
+		endforeach( source_file )
+		set( ${command_list} ${cmd_list} PARENT_SCOPE )
+	endfunction()
+
+	# Create list of commands for copying
+	set( WXWIDGETS_VERSION 310u )
```
Comment (AnotherFoxGuy) [inline source/main/CMakeLists.txt]: @only-a-ptr Could you change this to ```set( WXWIDGETS_VERSION 311u )```

Comment (only-a-ptr) [inline source/main/CMakeLists.txt]: done

File: source/main/CMakeLists.txt [hunk 2]
```diff
@@ -429,39 +374,233 @@ IF(WIN32)
...
+	set( WXWIDGETS_COMPILER vc140 )
```
Comment (only-a-ptr) [inline source/main/CMakeLists.txt]: done

**Comment by AnotherFoxGuy** [comment]
Now the only thing left to fix is Caelum

From the log:

	19:43:35: Parsing script RoRSkies.os
	19:43:35: Compiler error: unknown error in RoRSkies.os(1): token class, caelum_sky_system, unrecognized.
	...
	19:43:35:   ## Initializing Sky Subsystem
	19:43:35: Caelum: Initialising Caelum system...
	19:43:35: Caelum: Plugin not installed; installing now.
	19:43:35: Caelum plugin version 0.6.3 installed
	19:43:35: Registering ResourceManager for type PropertyScript
	19:43:35: Caelum: Creating required internal resource group 'Caelum'
	19:43:35: Creating resource group Caelum
	19:43:35: Caelum: Creating caelum sub-components.
	...
	19:43:35: Caelum: DONE initializing
	19:43:35: CaelumSystem: Attached to viewport 85787232 render target Rigs of Rods version 0.4.7.0-dev-0ae29fc
	19:43:35: OGRE EXCEPTION(6:FileNotFoundException): Cannot locate resource  in resource group Autodetect. in ResourceGroupManager::openResource at C:\projects\ror-dependencies\Source\ogre\OgreMain\src\OgreResourceGroupManager.cpp (line 761)
	19:43:35: exception upon loading sky script: OGRE EXCEPTION(6:FileNotFoundException): Cannot locate resource  in resource group Autodetect. in ResourceGroupManager::openResource at C:\projects\ror-dependencies\Source\ogre\OgreMain\src\OgreResourceGroupManager.cpp (line 761)

**Pull request merged**